flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherServices.java

```diff
@@ -74,22 +78,24 @@ public DispatcherServices(
 			@Nonnull GatewayRetriever<ResourceManagerGateway> resourceManagerGatewayRetriever,
```

Comment:
What is `@Nonnull` annotation conventions here? Following the statement

> This means that @Nonnull annotations are usually not necessary, but can be used in certain cases to override a previous annotation, or to point non-nullability out in a context where one would expect a nullable value.

I think here one would always expect a non-null value.

Reply:
Yes you are right. Since this change is not part of this PR I would like to address this problem as a follow up issue.